pipeline-steps/deploy.groovy

```diff
@@ -35,4 +35,62 @@ def deploy_sh(Map args) {
   ) // conditionalStage
 }
 
+def deploy_maas() {
+  common.conditionalStage(
+    stage_name: 'Prepare MaaS Configs',
+    stage: {
+      withCredentials([
+        string(
+          credentialsId: "dev_pubcloud_username",
+          variable: "PUBCLOUD_USERNAME"
+        ),
+        string(
+          credentialsId: "dev_pubcloud_api_key",
+          variable: "PUBCLOUD_API_KEY"
+        ),
+        string(
+          credentialsId: "dev_pubcloud_tenant_id",
+          variable: "PUBCLOUD_TENANT_ID"
+        )
+      ]){
+        sh """
+          set +x
+          export ADMIN_PWD=\$(sudo grep keystone_auth_admin_password /etc/openstack_deploy/user_osa_secrets.yml  | awk '{print \$2}')
+          set -x
+          export HOSTNAME=\$(hostname)
+
+          sudo -E sh -c 'echo "
+rackspace_cloud_username: ${env.PUBCLOUD_USERNAME}
```

Comment:
If we split gating_overrides into a few vars_files includes, could you then make use of the same mechanism? https://github.com/rcbops/rpc-gating/blob/master/playbooks/aio_config.yml That playbook could be generalised to config_overrides or similar.